### DISCUSSION TITLE
How not to behave

### DISCUSSION THREAD

**Discussion by azerik92** [bassamanator/Sovol-SV06-firmware#196] [General]
I've noticed a lot of the commands in this guide are inaccurate, and I've been consistently having to look up the correct commands in the klipper documentation.

**Comment by bassamanator** [comment]
Really? Let's see some examples.

**Reply by azerik92** [reply]
The section on PID tuning comes immediately to mind. 

` PID_TEST_BED TEMP=<temperature>` & `PID_TEST_HOTEND TEMP=<temperature>` both return an error that states the command is invalid, and seem to have been rolled into `PID_CALIBRATE HEATER=<config_name> TARGET=<temperature>`.

`DO_PROBE_CALIBRATE` is instead simply `PROBE_CALIBRATE`.

`DO_CREATE_MESH` is instead `BED_MESH_CALIBRATE`, which is incorrectly noted in Special Considerations as having changed from that to `_BED_MESH_CALIBRATE`.

**Reply by bassamanator** [reply]
Those are custom commands found only in this configuration.

If they're not working for you, then you're not using the klipper configuration found on this repo.

**Reply by azerik92** [reply]
> Those are custom commands found only in this configuration.

Then something is wrong with this configuration.

> If they're not working for you, then you're not using the klipper configuration found on this repo.

I can assure you I am, your username is plastered all over every config file. I can even see these custom commands listed in the CALIBRATION.cfg, but they didn't work, and returned as invalid commands as I mentioned. I followed this guide to the letter except for where these macros are concerned. 


More to the point, it seems unnecessarily complex to create macros that amount to [execute an existing command you could have just typed/pasted instead]. Even if you want to argue that they're half the character length and thus easier to type, that doesn't mean much when you can copy and paste it even faster than typing either command.

>`DO_CREATE_MESH` is instead `BED_MESH_CALIBRATE`, which is incorrectly noted in Special Considerations as having changed from that to `_BED_MESH_CALIBRATE.`

And this is still just blatantly wrong. The extra underscore causes it to return as invalid.

**Comment by bassamanator** [comment]
Here are my competitors. I assure you, their configurations are excellent! I recommend them to you, and to anyone else who doesn't want to use mine. And I'm sure there are other options as well.
![image](https://github.com/user-attachments/assets/4d9b29d8-14e9-41d1-af1f-445632a6d06b)

> Those are custom commands found only in this configuration. -me

> Then something is wrong with this configuration. -you

The Github star history would say otherwise. So would the [thank you messages](https://ko-fi.com/bassamanator) my users leave me. Here's one of my favourites:
![image](https://github.com/user-attachments/assets/e99f51db-f9fb-4a3b-a251-2acb22626e94)

> If they're not working for you, then you're not using the klipper configuration found on this repo. -me

> I can assure you I am, your username is plastered all over every config file. I can even see these custom commands listed in the CALIBRATION.cfg, but they didn't work, and returned as invalid commands as I mentioned. I followed this guide to the letter except for where these macros are concerned. -you

Linux is hard for a lot of people. I would recommend you ask a guardian who is tech savvy to help you with Klipperizing your 3D printer (see the note above about a parent helping their 9-year-old with this process). If you don't have such a guardian, then I would suggest you ask a friend from school who is good with computers to help you.

If you still can't find someone to help you, then I recommend hiring someone from upwork or fiverr. There is no shame in paying someone to do something for you, something that you are unable to do yourself. And it won't be expensive at all, I bet you can find someone who is good with Linux to help you for like $10, if not less. In fact, if you can prove you hired someone to help, `I will cover half your cost and donate $5 to you myself, how's that?`

And if you had a better attitude, someone from the community would be happy to help you. There are thousands of us who spend our time writing documentation, creating repos, writing code, answering questions, helping others basically, `completely for free`.

> More to the point, it seems unnecessarily complex to create macros that amount to [execute an existing command you could have just typed/pasted instead]. Even if you want to argue that they're half the character length and thus easier to type, that doesn't mean much when you can copy and paste it even faster than typing either command. -you

Those macros are doing a lot more than that, but you've successfully demonstrated that you wouldn't understand, so I won't bother explaining, but feel free to read the code at your leisure. You can ask chatgpt to explain everything to you.

> And this is still just blatantly wrong. The extra underscore causes it to return as invalid. -you

Just to make sure that nothing went wrong with Github (which has happened before), I wiped out my `~/printer_data/config`, then followed the instructions to download, and I had an error free Mainsail dashboard in less than 1 minute. I also ran the command that you are so obsessed with, `_BED_MESH_CALIBRATE`, and it works exactly as it should.

**Reply by azerik92** [reply]
>And if you had a better attitude, someone from the community would be happy to help you. There are thousands of us who spend our time writing documentation, creating repos, writing code, answering questions, helping others basically, `completely for free`.

Dude calm down, I don't know what attitude you think I have, all I did was report errors I received. I already got everything working by looking up the commands in the Klipper documentation and doing what your macros were supposed to do manually, and was simply letting you know that something from the guide wasn't working on my end. 

Edge cases happen, you don't need to take it personally and throw a fit when a single person doesn't have a perfect experience with your work. You asked me what commands didn't work, I tell you in good faith what returned errors for me, and your response is to demean and insult me? What I found in your configuration files clashes with what I find in Klipper documentation, and spat errors out. Maybe something is wrong with my CB1, maybe something somewhere didn't get downloaded. I can't tell you what went wrong, only that something DID go wrong.

If you can't reproduce the issue, or can't be bothered to, oh well for me, I already got around it, I was just trying to be a good samaritan and report my issues on the chance they happen to anyone else. 

And here's a friendly word of advice

>The Github star history would say otherwise. So would the [thank you messages](https://ko-fi.com/bassamanator) my users leave me. Here's one of my favourites:

Getting a big head over internet points makes any inference of childishness just disgustingly hypocritical. Maybe I would have been one of those thank you donations if you hadn't jumped down my throat like I pissed in your cereal over a bug report.

Don't know why I bothered, That code of conduct section on this repo is a bad joke it seems.
